clients/drcachesim/tools/func_view.cpp

```diff
@@ -258,12 +278,15 @@ func_view_t::process_memref(const memref_t &memref)
         break;
     }
     case TRACE_MARKER_TYPE_FUNC_RETVAL: {
-        --shard->nesting_level;
-        std::string indent;
+        if (shard->nesting_level > 0) {
+            --shard->nesting_level;
+        } else {
+            std::cerr << "WARNING: Unnested FUNC_RETVAL\n";
```

Comment:
"Unnested" is confusing here: really the "nesting" we want is just at least one call entry.  How about "WARNIG: FUNC_RETVAL found without a function entry" or something.

Reply:
I don't know the full marker semantics, but the code only increments nesting_level when it sees RETADDR markers, so I think it's accurate to warn of a RETVAL without prior RETADDR.

Reply:
That is reasonable.